# Statistical Analysis Experiment for Depression

This notebook will produce the results for the statistical analysis experiment of my final year project "<i>Statistical Analysis for Detection Depression</i>". The statistical analysis method used for this project is descriptive analysis.<br><br>

The descriptive analysis technique has two types which are <mark><b>central tendency</b></mark> and <mark><b>variability</b></mark>. For this project, it will use the central tendency methods: mean, median, mode, maximum, and minimum and variability methods: standard deviation and variance.<br><br>

Deliverable requirements for statistical analysis experiment:
- [ ] Use descriptive analysis on the MADRS score data to analyse the central tendency and variability of the number of days all patients have measured
- [ ] In the MADRS score data, separate into two dataframes for depressed and non-depressed patients using descriptive analysis to analyse the central tendency and variability of the number of days they measured
- [ ] Calculate the difference in the MADRS score for the depressed patients
- [ ] Use descriptive analysis to analyse the MADRS score at the start and end of measurement and its difference for patients with depression
- [ ] Use data visualisation to show the MADRS score for all depressed patients
- [ ] Use data visualisation to show the number of days that depressed and non-depressed patients have measured

## Initial stages

Firstly, import the relevant libraries and load the <code>scores.csv</code> dataframe using Pandas and display the data.

In [1]:
# Import the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
# Load the scores.csv file and display the data
scores_df = pd.read_csv("data/scores.csv")
scores_df

number  days  gender    age  afftype  melanch  inpatient    edu  \
0    condition_1    11       2  35-39      2.0      2.0        2.0   6-10   
1    condition_2    18       2  40-44      1.0      2.0        2.0   6-10   
2    condition_3    13       1  45-49      2.0      2.0        2.0   6-10   
3    condition_4    13       2  25-29      2.0      2.0        2.0  11-15   
4    condition_5    13       2  50-54      2.0      2.0        2.0  11-15   
5    condition_6     7       1  35-39      2.0      2.0        2.0   6-10   
6    condition_7    11       1  20-24      1.0      NaN        2.0  11-15   
7    condition_8     5       2  25-29      2.0      NaN        2.0  11-15   
8    condition_9    13       2  45-49      1.0      NaN        2.0   6-10   
9   condition_10     9       2  45-49      2.0      2.0        2.0   6-10   
10  condition_11    14       1  45-49      2.0      2.0        2.0   6-10   
11  condition_12    12       2  40-44      1.0      2.0        2.0   6-10   
12  condition_13    14       2  35-39      1.0      2.0        2.0  11-15   
13  condition_14    14       1  60-64      1.0      2.0        2.0   6-10   
14  condition_15    13       2  55-59      2.0      2.0        2.0  11-15   
15  condition_16    16       1  45-49      2.0      2.0        2.0  11-15   
16  condition_17    13       1  50-54      1.0      2.0        2.0   6-10   
17  condition_18    13       2  40-44      3.0      2.0        2.0  11-15   
18  condition_19    13       2  50-54      2.0      2.0        1.0  16-20   
19  condition_20    13       1  30-34      2.0      1.0        1.0   6-10   
20  condition_21    13       2  35-39      2.0      2.0        1.0   6-10   
21  condition_22    14       1  65-69      2.0      2.0        1.0          
22  condition_23    16       1  30-34      2.0      2.0        1.0  16-20   
23     control_1     8       2  25-29      NaN      NaN        NaN    NaN   
24     control_2    20       1  30-34      NaN      NaN        NaN          
25     control_3    12       2  30-34      NaN      NaN        NaN          
26     control_4    13       1  25-29      NaN      NaN        NaN          
27     control_5    13       1  30-34      NaN      NaN        NaN          
28     control_6    13       1  25-29      NaN      NaN        NaN          
29     control_7    13       1  20-24      NaN      NaN        NaN          
30     control_8    13       2  40-44      NaN      NaN        NaN          
31     control_9    13       2  30-34      NaN      NaN        NaN          
32    control_10     8       1  30-34      NaN      NaN        NaN          
33    control_11    13       1  45-49      NaN      NaN        NaN          
34    control_12    14       1  60-64      NaN      NaN        NaN          
35    control_13    13       1  50-54      NaN      NaN        NaN    NaN   
36    control_14    13       1  50-54      NaN      NaN        NaN          
37    control_15    11       1  45-49      NaN      NaN        NaN          
38    control_16    13       2  40-44      NaN      NaN        NaN          
39    control_17     9       1  45-49      NaN      NaN        NaN          
40    control_18    13       2  20-24      NaN      NaN        NaN          
41    control_19    13       1  50-54      NaN      NaN        NaN          
42    control_20    13       1  35-39      NaN      NaN        NaN          
43    control_21     8       1  50-54      NaN      NaN        NaN          
44    control_22    13       1  25-29      NaN      NaN        NaN          
45    control_23    13       1  20-24      NaN      NaN        NaN          
46    control_24    13       2  20-24      NaN      NaN        NaN          
47    control_25    13       1  65-69      NaN      NaN        NaN          
48    control_26    13       1  35-39      NaN      NaN        NaN          
49    control_27    13       2  50-54      NaN      NaN        NaN          
50    control_28    16       2  45-49      NaN      NaN        NaN          
51    

In [3]:
# Display the first 5 rows
scores_df.head()

number  days  gender    age  afftype  melanch  inpatient    edu  \
0  condition_1    11       2  35-39      2.0      2.0        2.0   6-10   
1  condition_2    18       2  40-44      1.0      2.0        2.0   6-10   
2  condition_3    13       1  45-49      2.0      2.0        2.0   6-10   
3  condition_4    13       2  25-29      2.0      2.0        2.0  11-15   
4  condition_5    13       2  50-54      2.0      2.0        2.0  11-15   

   marriage  work  madrs1  madrs2  
0       1.0   2.0    19.0    19.0  
1       2.0   2.0    24.0    11.0  
2       2.0   2.0    24.0    25.0  
3       1.0   1.0    20.0    16.0  
4       2.0   2.0    26.0    26.0

In [4]:
# Display the last 5 rows
scores_df.tail()

number  days  gender    age  afftype  melanch  inpatient edu  \
50  control_28    16       2  45-49      NaN      NaN        NaN       
51  control_29    13       2  50-54      NaN      NaN        NaN       
52  control_30     9       2  35-39      NaN      NaN        NaN       
53  control_31    13       1  20-24      NaN      NaN        NaN       
54  control_32    14       2  25-29      NaN      NaN        NaN       

    marriage  work  madrs1  madrs2  
50       NaN   NaN     NaN     NaN  
51       NaN   NaN     NaN     NaN  
52       NaN   NaN     NaN     NaN  
53       NaN   NaN     NaN     NaN  
54       NaN   NaN     NaN     NaN

In the <code>scores.csv</code> file, there are 55 patients in total. 23 patients are depressed (labelled <mark><b>condition</b></mark>) and 32 patients are non-depressed (labelled <mark><b>control</b></mark>).

In [5]:
scores_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55 entries, 0 to 54
Data columns (total 12 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   number     55 non-null     object 
 1   days       55 non-null     int64  
 2   gender     55 non-null     int64  
 3   age        55 non-null     object 
 4   afftype    23 non-null     float64
 5   melanch    20 non-null     float64
 6   inpatient  23 non-null     float64
 7   edu        53 non-null     object 
 8   marriage   23 non-null     float64
 9   work       23 non-null     float64
 10  madrs1     23 non-null     float64
 11  madrs2     23 non-null     float64
dtypes: float64(7), int64(2), object(3)
memory usage: 5.3+ KB


As shown in the information above, majority of the columns have missing values but for the statistical experiment, the days, madrs1 and madrs2 columns are the main focus. We want to know the central tendency and variability for the number of days patients have measured and the MADRS scores for depressed patients in order to retrieve the basis of the data.

## Central Tendency and Variability for the number of days all patients measured their activity

For this statistical analysis experiment, I will use the descriptive analysis to analyse the central tendency and variability for the number of days all patients have measured.<br>

In this deliverable, the <code>days</code> column is the column we are focusing as we want to know the mean, mode, median, minimum, maximum, standard deviation and variance for all patients.

In [12]:
'''
Index 0 is condition_1 to index 23 is condition_22
Index 23 is control_1 to index 54 is control_32
'''
# First look at the number of days all patients measure their activity
scores_df["days"]

0     11
1     18
2     13
3     13
4     13
5      7
6     11
7      5
8     13
9      9
10    14
11    12
12    14
13    14
14    13
15    16
16    13
17    13
18    13
19    13
20    13
21    14
22    16
23     8
24    20
25    12
26    13
27    13
28    13
29    13
30    13
31    13
32     8
33    13
34    14
35    13
36    13
37    11
38    13
39     9
40    13
41    13
42    13
43     8
44    13
45    13
46    13
47    13
48    13
49    13
50    16
51    13
52     9
53    13
54    14
Name: days, dtype: int64

In [13]:
# Count of the number of days all patients measure their activity
scores_df["days"].value_counts()

13    31
14     6
11     3
9      3
16     3
8      3
12     2
18     1
7      1
5      1
20     1
Name: days, dtype: int64

In [14]:
# Mean (average) number of days of all patients
scores_df["days"].mean()

12.6

In [15]:
# Mode (most common) number of days of all patients
scores_df["days"].mode()

0    13
Name: days, dtype: int64

In [16]:
# Median (middle) number of days of all patients
scores_df["days"].median()

13.0

In [17]:
# Minimum number of days of all patients
scores_df["days"].min()

5

In [18]:
# Maximum number of days of all patients
scores_df["days"].max()

20

In [19]:
# Standard deviation of number of days of all patients
scores_df["days"].std()

2.4914669187864833

In [20]:
# Variance of number of days of all patients
scores_df["days"].var()

6.2074074074074135

To put into conclusion for this deliverable, the results for the number of days all patients have measured their activity are:
* Mean = 12.6
* Mode = 13
* Median = 13
* Minimum = 5
* Maximum = 20
* Standard deviation = 2.4914669187864833 (approx. 2.5)
* Variance = 6.2074074074074135 (approx. 6.2)